bouncingbullet/assets/js/game/controls.js

```diff
@@ -0,0 +1,21 @@
+import join_channel from './socket';
+
+class Controls {
+  constructor() {
+    this.channel = join_channel("input");
+    window.addEventListener("keyup", this.keyhandler.bind(this), true);
+  }
+  keyhandler(e) {
+    let valid = [32, 37, 38, 39, 40, 13].includes(e.keyCode);
```

Comment:
Not a criticism, but a suggestion:

In general, returns in conditionals can be confusing to the poor reader. This is exacerbated if the conditional has an `else` clause, because that implies that execution will continue after the `if` regardless.

When I feel I _have_ to do it, I'd lay it out like:

~~~ elixir
if (!valid)
  return;

console.log(e.key);
this.channel.push(e.key);

e.preventDefault();
~~~

If you lay it out this way, it then becomes apparent that the `preventDefault` call won't happen for the invalid case. This may be the desired effect, but if so having it outside the `if` in the original code made that really unclear.